pkg/runtime/reconciler.go

```diff
@@ -268,49 +268,113 @@ func (r *resourceReconciler) Sync(
 					// there is some changes available in the latest.RuntimeObject()
 					// (example: ko.Status.Conditions) which have been
 					// updated in the resource
-					// Thus, patchResource() call should be made here
-					_ = r.patchResource(ctx, desired, latest)
+					// Thus, patchResourceStatus() call should be made here
+					_ = r.patchResourceStatus(ctx, desired, latest)
 				}
 				return err
 			}
 			rlog.Info("updated resource")
 		}
 	}
-	err = r.patchResource(ctx, desired, latest)
-	if err != nil {
-		return err
-	}
-	for _, condition := range latest.Conditions() {
-		if condition.Type == ackv1alpha1.ConditionTypeResourceSynced {
-			if condition.Status == corev1.ConditionTrue {
-				if duration := r.rmf.RequeueOnSuccessSeconds(); duration > 0 {
+	if ackcompare.IsNotNil(latest) {
```

Comment:
If `err` != `nil` and `latest` is `nil`, we will no longer call `patchResource`. Is that what you meant?

Reply:
It's just a safety nil check before calling patchResource method so we do not try to patch from nil object or read conditions from nil object.
And yes, if latest is nil, we do not call patchResource method.